src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/Dac/DacSymbolsUtils.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Acuminator.Utilities.Common;
+using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Acuminator.Utilities.Roslyn.Semantic.Dac
+{
+    public static class DacSymbolsUtils
+    {
+        public static IEnumerable<ITypeSymbol> GetDacExtensionsWithDac(this ITypeSymbol dacExtension, PXContext pxContext)
+        {
+            dacExtension.ThrowOnNull(nameof(dacExtension));
+            pxContext.ThrowOnNull(nameof(pxContext));
+
+            if (!dacExtension.InheritsFrom(pxContext.PXCacheExtensionType))
+            {
+                return Enumerable.Empty<ITypeSymbol>();
+            }
+
+            var extensionBaseType = dacExtension.BaseType;
+            var typeArguments = extensionBaseType.TypeArguments;
+            var dacType = typeArguments.LastOrDefault();
+
+            if (dacType == null || !dacType.IsDac(pxContext))
+            {
+                return Enumerable.Empty<ITypeSymbol>();
+            }
+
+            var types = new List<ITypeSymbol>() { dacExtension };
```

Comment:
the estimated capacity for the List is already known - it is `typeArguments.Length - 1`. You can pass it to the List constructor to improve perfomance